### PR TITLE
[11.0][IMP] product_sequence: Remove default_code sql constraint

### PR DESCRIPTION
**Pull request by sergio-teruel**
The sql restriction is added in product_code_unique, so if you need this restriction, please install this module.

cc @Tecnativa

### CODE REVIEW

File: product_sequence/README.rst [hunk 1]
```diff
@@ -1,17 +1,41 @@
...
+    :target: https://translation.odoo-community.org/projects/product-attribute-11-0/product-attribute-11-0-product_sequence
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/135/11.0
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
 This module allows to associate a sequence to the product reference.
 The reference (default code) is unique (SQL constraint) and required.
```
Comment (pedrobaeza) [inline product_sequence/README.rst]: Change this text for advising the installation of the other module.

File: product_sequence/readme/INSTALL.rst [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Prior to installing this module, if you have any existing products you should
```
Comment (pedrobaeza) [inline product_sequence/readme/INSTALL.rst]: This should be removed also.

**Comment by sergio-teruel** [comment]
@pedrobaeza changes done!

File: product_sequence/readme/INSTALL.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Products with a default_code of '/' or empty will automatically be assigned a
```
Comment (pedrobaeza) [inline product_sequence/readme/INSTALL.rst:1]: Is this mechanism still needed now that the constraint has been removed?

**Comment by pedrobaeza** [comment]
Sorry to insist, but I still don't see the reference to install the other module as said.

**Comment by Cedric-Pigeon** [comment]
@sergio-teruel Would not it be better to put product_code_unique in dependencies to replace the sql constrains. Removing the constrains as this will change the behaviour....

**Review by Cedric-Pigeon — CHANGES_REQUESTED** [review]
See above

**Comment by pedrobaeza** [comment]
But the idea is to not break existing installations adding a dependency. Moreover, we are doing this because we don't want that constraint. It's not the aim of the module, and for our use case, it's wasting our customer DB.

**Comment by Cedric-Pigeon** [comment]
@pedrobaeza This addon aims to link a sequence to the product default code. I don't see any cases in which a sequence can provide duplicates.

**Comment by pedrobaeza** [comment]
There are several, like obsolescence of certain products that are replaced by others with the same internal code, mistakes with the product UoM that requires to create a new product, but they want to have both with the code for searching, etc.

Any way, this module shouldn't extra-limit its purpose. You have the other one for the constraint.

**Comment by Cedric-Pigeon** [comment]
@pedrobaeza Indeed. I agree. Than it would be useful to add a history section in the readme to explain the change and to let people who will migrate from previous versions what they have to do to keep the constrain

**Comment by pedrobaeza** [comment]
Well, that's why I was insisting to @sergio-teruel to put that information in the README. Put it as HISTORY can be other option indeed.

**Comment by sergio-teruel** [comment]
@pedrobaeza @Cedric-Pigeon changes done

**Review by pedrobaeza — APPROVED** [review]

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: LGTM + Tested

**Pull request merged**